app/code/Magento/OfflinePayments/Observer/BeforeOrderPaymentSaveObserver.php

```diff
@@ -33,15 +35,19 @@ public function execute(\Magento\Framework\Event\Observer $observer)
         if (in_array($payment->getMethod(), $instructionMethods)) {
             $payment->setAdditionalInformation(
                 'instructions',
-                $payment->getMethodInstance()->getInstructions()
+                $payment->getMethodInstance()->getConfigData(
+                    'instructions',
+                    $payment->getOrder()->getStoreId()
+                )
             );
         } elseif ($payment->getMethod() === Checkmo::PAYMENT_METHOD_CHECKMO_CODE) {
             $methodInstance = $payment->getMethodInstance();
-            if (!empty($methodInstance->getPayableTo())) {
-                $payment->setAdditionalInformation('payable_to', $methodInstance->getPayableTo());
+            $storeId = $payment->getOrder()->getStoreId();
+            if (!empty($payableTo = $methodInstance->getConfigData('payable_to', $storeId))) {
```

Comment:
I also used to implement assignments within comparisons but one day someone pointed me out that it can be a bit misleading: is the `=` a real assignment or a bug, that is, a missing`==`? Reading the code without too much attention (it happens) it may seem a bug, someone could be tempted to correct it causing indeed a bug :) 

At that point, I stopped using it and switched back to assigning the variable and doing the comparison in two different statements.

Would you consider doing that change, thus joining my group of old, wise, and defensive programmers like me? :-)

Reply:
@aleron75 I'm not sure if I will get used to do it, but I can definitely try, changed it :)